tests/test_example.py

```diff
@@ -0,0 +1,43 @@
+from attrs import define
+
+from web_poet import ItemPage, default_registry, field, handle_urls
+from web_poet.example import get_item
+
+
+def test_async_to_item():
+    @define
+    class Item:
+        foo: str
+
+    try:
+
+        @handle_urls("")
+        class Page(ItemPage[Item]):
+            @field
+            async def foo(self):
+                return "bar"
+
+        item = get_item("file:///dev/null", Item)
+        assert isinstance(item, Item)
+        assert item.foo == "bar"
+    finally:
+        default_registry._rules.pop()
```

Comment:
We'll need to update this `RulesRegistry._rules` has became a `dict`. https://github.com/scrapinghub/web-poet/pull/112/files#diff-22f478b54a07b372ea47ead887ca01488362ad501692903a9d303511f9d1fe24L118